Sprint-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,63 @@
 const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+  "./assets/cute-cat-c.jpg",
 ];
 
+// Write your code here
 
-// Write your code here
\ No newline at end of file
+let currentIndex = 0;
+const carouselImg = document.getElementById("carousel-img");
+let slideshowInterval = null;
+const imageCountText = document.getElementById("image-count");
+
+const autoForwardBtn = document.getElementById("auto-forward");
+const autoBackwardBtn = document.getElementById("auto-backward");
+const stopBtn = document.getElementById("stop");
+
+function updateImage() {
+  carouselImg.setAttribute("src", images[currentIndex]);
+  imageCountText.textContent = `Image ${currentIndex + 1} of ${images.length}`;
+}
+
+function goForward() {
+  currentIndex = (currentIndex + 1) % images.length;
+  updateImage();
+}
+
+function goBackward() {
+  currentIndex = (currentIndex - 1 + images.length) % images.length;
+  updateImage();
+}
+
+document.getElementById("forward-btn").addEventListener("click", goForward);
+document.getElementById("backward-btn").addEventListener("click", goBackward);
+
+// Auto Forward logic
+function startAutoForward() {
+  stopSlideshow();
+  slideshowInterval = setInterval(goForward, 2000);
+  autoForwardBtn.disabled = true;
+  autoBackwardBtn.disabled = true;
+}
+
+// Auto Backward logic
+function startAutoBackward() {
+  stopSlideshow();
+  slideshowInterval = setInterval(goBackward, 2000);
+  autoForwardBtn.disabled = true;
+  autoBackwardBtn.disabled = true;
+}
```

Comment:
These two functions seem to mostly do the same thing.

Imagine if we decided we also wanted to disable the "Forward" and "Backward" buttons when in auto mode, and _only_ allow the "Stop" button to be pressed. We would need to make the same change to both of these functions. Maybe we wouldn't remember this, and would only change one, and have a bug.

Can you think of a way to avoid this, so that we re-use this code across both functions, rather than having two copies of it?